network/p2p/p2p.go

```diff
@@ -180,18 +181,24 @@ func configureResourceManager(cfg config.Local) (network.ResourceManager, error)
 	return rm, err
 }
 
+// StreamHandlerMap is a map of protocol IDs to StreamHandlers
+type StreamHandlerMap map[protocol.ID]StreamHandler
+
 // MakeService creates a P2P service instance
-func MakeService(ctx context.Context, log logging.Logger, cfg config.Local, h host.Host, listenAddr string, wsStreamHandler StreamHandler, metricsTracer pubsub.RawTracer) (*serviceImpl, error) {
+func MakeService(ctx context.Context, log logging.Logger, cfg config.Local, h host.Host, listenAddr string, wsStreamHandlers StreamHandlerMap, metricsTracer pubsub.RawTracer) (*serviceImpl, error) {
 
-	sm := makeStreamManager(ctx, log, h, wsStreamHandler, cfg.EnableGossipService)
+	sm := makeStreamManager(ctx, log, h, wsStreamHandlers, cfg.EnableGossipService)
+	sub, err := h.EventBus().Subscribe(new(event.EvtPeerIdentificationCompleted))
+	if err != nil {
+		err0 := fmt.Errorf("failed to subscribe to peer identification events: %v", err)
+		return nil, err0
+	}
+	go sm.peerWatcher(ctx, sub)
 	h.Network().Notify(sm)
-	h.SetStreamHandler(AlgorandWsProtocol, sm.streamHandler)
 
-	// set an empty handler for telemetryID/telemetryInstance protocol in order to allow other peers to know our telemetryID
```

Comment:
For nodes that currently have P2P enabled that do not take this upgrade right away, what will be the consequences of dropping the telemetryID for them (when connecting to upgraded nodes)?

Reply:
in some telemetry messages there will be empty field for other node.